graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java

```diff
@@ -110,16 +101,26 @@ public Set<NodeFileDescriptorStats> fileDescriptorStats() {
     }
 
     private List<NodeResponse> nodes() {
-        final List<NodeResponse> allNodes = catApi.nodes();
-        final List<NodeResponse> nodesWithDiskStatistics = allNodes.stream().filter(NodeResponse::hasDiskStatistics).collect(Collectors.toList());
+        final var response = client.execute(c -> c.cat().nodes(builder -> builder.fullId(true)
+                .headers("id,name,role,host,ip,fileDescriptorMax,diskUsed,diskTotal,diskUsedPercent")));
+        final var allNodes = response.valueBody()
+                .stream()
+                .map(this::createNodeResponse)
+                .toList();
+        final var nodesWithDiskStatistics = allNodes.stream().filter(NodeResponse::hasDiskStatistics).toList();
```

Comment:
How about using a groupingBy Collector to split the stream in one step?

Reply:
Sure. I think that should go into a follow-up PR though.